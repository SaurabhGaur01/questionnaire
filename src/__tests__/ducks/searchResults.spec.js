import {
    setSearchResults,
    clearSearchResults,
    searchResultsReducer,
} from '../../ducks/searchResults';

const mockObject = [
    {
        mockVariable: 'mockValue',
    }
];

describe('searchResults Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setSearchResults should match the snapshot' , () => {
            expect(setSearchResults(mockObject)).toMatchSnapshot();
        });
        it('clearSearchResults should match the snapshot' , () => {
            expect(clearSearchResults()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(searchResultsReducer()).toEqual([]);
        });
        it('should return the passed state if called with no action' , () => {
            expect(searchResultsReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(searchResultsReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value of the API call' , () => {
            expect(searchResultsReducer('', setSearchResults(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearSearchResults' , () => {
            expect(searchResultsReducer(mockObject, clearSearchResults())).toEqual([]);
        });
    });
});