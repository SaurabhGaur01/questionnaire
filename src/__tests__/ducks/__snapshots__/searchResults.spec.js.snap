// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchResults Duck Tests Action Generator clearSearchResults should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/CLEAR_SEARCH_RESULTS",
}
`;

exports[`searchResults Duck Tests Action Generator setSearchResults should match the snapshot 1`] = `
Object {
  "data": Array [
    Object {
      "mockVariable": "mockValue",
    },
  ],
  "type": "modules/questionnaire-app/SET_SEARCH_RESULTS",
}
`;
